artifactory/services/buildinfo.go

```diff
@@ -0,0 +1,68 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/artifactory/auth"
+	"github.com/jfrog/jfrog-client-go/artifactory/buildinfo"
+	rthttpclient "github.com/jfrog/jfrog-client-go/artifactory/httpclient"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+)
+
+type BuildInfoService struct {
+	client     *rthttpclient.ArtifactoryHttpClient
+	ArtDetails auth.ArtifactoryDetails
+}
+
+func NewBuildInfoService(client *rthttpclient.ArtifactoryHttpClient) *BuildInfoService {
+	return &BuildInfoService{client: client}
+}
+
+func (bis *BuildInfoService) GetArtifactoryDetails() auth.ArtifactoryDetails {
+	return bis.ArtDetails
+}
+
+func (bis *BuildInfoService) SetArtifactoryDetails(rt auth.ArtifactoryDetails) {
+	bis.ArtDetails = rt
+}
+
+func (bis *BuildInfoService) GetJfrogHttpClient() (*rthttpclient.ArtifactoryHttpClient, error) {
```

Comment:
The error returned value seems redundant.

Reply:
The BuildInfoService struct implements the CommonConf interface so this is mandatory.